app/code/Magento/Cms/Test/Mftf/ActionGroup/FillOutBlockContentActionGroup.xml

```diff
@@ -18,4 +18,14 @@
         <selectOption selector="{{BlockNewPageBasicFieldsSection.storeView}}" userInput="All Store View" stepKey="selectAllStoreView"/>
         <fillField selector="{{BlockContentSection.TextArea}}" userInput="{{_defaultBlock.content}}" stepKey="fillContentField"/>
     </actionGroup>
+    <actionGroup name="FillOutBlockContentWithMarginalSpace">
```

Comment:
Please, put this action group in a separate file, rename it as `AdminFillCmsBlockFormActionGroup`. Make this action group accept 3 arguments `identifier`, `title` and `content` and pass the corresponding arguments from your test. 

After this change, you will not need a separate `BlockDataWithMarginalSpace` entity created in `app/code/Magento/Cms/Test/Mftf/Data/BlockPageData.xml`.

Unfortunately, the existing MFTF tests in the core are not the best example of best practices for writing MFTF test. 

Thank you!

Reply:
@rogyar Thanks, for the suggestion to improve the code quality. I have made changes to the code as per your suggestions. Please review it.
Please let me know the best practices in MFTF, so that I will also refactor the existing codes in future.